src/main/scala/workflow/AutoCacheRule.scala

```diff
@@ -43,12 +43,17 @@ class AutoCacheRule(cachingMode: CachingStrategy) extends Rule with Logging {
    * Get an estimate for how many times each instruction will be executed, assuming
    * the given set of instructions have their outputs cached
    */
-  def getRuns(instructions: Seq[Instruction], cache: Set[Int], nodeWeights: Map[Int, Int]): Map[Int, Int] = {
-    val immediateChildrenByInstruction = getImmediateChildrenByInstruction(instructions)
```

Comment:
So it looks like pre-computing this value ahead of time was the main source of savings. Is that right? Out of curiousity did you find this by profiling?